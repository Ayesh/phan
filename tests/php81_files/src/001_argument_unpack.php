<?php

function foo(int $bar) {
    var_dump($bar);
}
$args = ['bar' => 123];
foo(...$args);  // Phan should NOT warn because 'minimum_target_php_version' is 8.1 (8.0+)
$other = ['bar' => 'invalid'];
foo(...$other);  // Phan should warn because the type is definitely invalid (TODO: improve)
