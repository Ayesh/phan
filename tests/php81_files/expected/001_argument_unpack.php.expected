%s:9 PhanTypeMismatchArgumentReal Argument 1 ($bar) is ...($other) of type 'invalid' but \foo() takes int defined at %s:3
